calico-vpp-agent/routing/bgp_watcher.go

```diff
@@ -333,28 +510,55 @@ func (w *Server) WatchBGPPath(t *tomb.Tomb) error {
 					return err
 				}
 			case common.BGPPeerAdded:
-				peer := evt.New.(*bgpapi.Peer)
+				localPeer := evt.New.(*watchers.LocalBGPPeer)
+				peer := localPeer.Peer
+				filters := localPeer.BGPFilterNames
+				BGPPolicies, err := w.filterPeer(peer.Conf.NeighborAddress, filters)
```

Comment:
We can probably not return `BGPPolicies` nor store it in `localPeer.BGPPolicies` as this is just a computation from `w.peers` and `w.filters` right ?

Reply:
The stored policies are there for deletion. I think the policy name is insufficient. The api is unclear though so not sure.

Reply:
Right, my bad. I got mixed up.